src/components/implementation/tests/unit_cxx/cxx.h

```diff
@@ -0,0 +1,209 @@
+#ifndef CXX_UNIT_H
+#define CXX_UNIT_H
+
+#include <vector>
+#include <algorithm>
+#include <set>
+#include <queue>
+#include <map>
+#include <string>
+#include <iostream>
+using namespace std;
+#ifdef __cplusplus
+extern "C" {
+#include <print.h>
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+class Global_foo
+{
+public:
+	Global_foo();
+	~Global_foo();
+};
+
+Global_foo::Global_foo()
+{
+	printc("Global foo constructor\n");
```

Comment:
It seems that `cout` is not initialized when calling the constructors. I don't know how to fix this.